File: go/flatten_directory.go
// flatten_directory.go
package main

import (
    "fmt"
    "io"
    "os"
    "path/filepath"
)

func flattenDirectory(sourceDir, targetDir string) {
    if _, err := os.Stat(sourceDir); os.IsNotExist(err) {
        fmt.Println("Error: Source directory does not exist.")
        os.Exit(1)
    }

    os.MkdirAll(targetDir, os.ModePerm)

    filepath.Walk(sourceDir, func(path string, info os.FileInfo, err error) error {
        if err != nil {
            return err
        }
        if !info.IsDir() {
            targetPath := filepath.Join(targetDir, info.Name())
            copyFile(path, targetPath)
        }
        return nil
    })
    fmt.Printf("All files have been extracted to %s without preserving folder structure.\n", targetDir)
}

func copyFile(source, target string) {
    sourceFile, err := os.Open(source)
    if err != nil {
        fmt.Printf("Error opening source file %s: %v\n", source, err)
        return
    }
    defer sourceFile.Close()

    targetFile, err := os.Create(target)
    if err != nil {
        fmt.Printf("Error creating target file %s: %v\n", target, err)
        return
    }
    defer targetFile.Close()

    _, err = io.Copy(targetFile, sourceFile)
    if err != nil {
        fmt.Printf("Error copying file %s to %s: %v\n", source, target, err)
    }
}

func main() {
    if len(os.Args) != 3 {
        fmt.Println("Usage: flatten_directory <source_directory> <target_directory>")
        os.Exit(1)
    }
    flattenDirectory(os.Args[1], os.Args[2])
}

File: python/flatten_directory.py
# flatten_directory.py
import os
import shutil
import sys

def flatten_directory(source_dir: str, target_dir: str):
    if not os.path.exists(source_dir):
        print("Error: Source directory does not exist.")
        sys.exit(1)

    os.makedirs(target_dir, exist_ok=True)

    for root, _, files in os.walk(source_dir):
        for file in files:
            shutil.copy2(os.path.join(root, file), os.path.join(target_dir, file))

    print(f"All files have been extracted to {target_dir} without preserving folder structure.")

if __name__ == "__main__":
    if len(sys.argv) != 3:
        print("Usage: python flatten_directory.py <source_directory> <target_directory>")
        sys.exit(1)

    flatten_directory(sys.argv[1], sys.argv[2])
